cli/src/merge_tools/mod.rs

```diff
@@ -179,10 +181,18 @@ impl DiffEditTool {
         settings: &UserSettings,
         name: &str,
     ) -> Result<Option<Self>, MergeToolConfigError> {
-        match name {
-            BUILTIN_EDITOR_NAME => Ok(Some(DiffEditTool::Builtin)),
-            _ => Ok(get_external_tool_config(settings, name)?.map(DiffEditTool::external)),
+        if let BUILTIN_EDITOR_NAME = name {
+            return Ok(Some(DiffEditTool::Builtin));
+        }
+        let maybe_tool = get_external_tool_config(settings, name)?;
+        if let Some(tool) = &maybe_tool {
+            if tool.edit_args.is_empty() {
+                return Err(MergeToolConfigError::EditArgsNotConfigured {
+                    tool_name: name.to_string(),
+                });
+            }
         }
```

Comment:
I think it's better to check bad configuration by `DiffEditor::new_inner()` as we do for `MergeEditor`.

Reply:
In particular, AFAICT there is currently no case when `&args` is passed to `MergeEditor::new_inner` as `name`, and the `name` is actually used.

Reply:
Maybe we could move handling of `::with_program(name)` to `::get_tool_config()`, but I have no idea if it's better to add separate check for `ui.merge-editor = ["command-name-without-args"]`. Beware that merge editors don't have a predefined `merge_args`.

`DiffEditor` doesn't have this problem, but I would just copy the structure of `MergeEditor`.

Reply:
I mean `ExternalMergeTool` is constructed in various ways, and some/all of them should be checked if `args` is empty or not. `::get_tool_config()` is just one example.

Reply:
First of all, are you happy with the way the code is now?

I think I see what you're saying, what I said in https://github.com/jj-vcs/jj/pull/7009#discussion_r2217034130 is not right in case the user has `ui.merge-tool=[]`. I'd address it by adding the check to `MergeTool::external` and removing the `unwrap_or_else` from `MergeEditor::with_name`, since I think that codepath never results in a valid tool.

One of us could try to refactor it later, or not.

Reply:
The one I last checked looked good to me, thanks.

> in case the user has `ui.merge-tool=[]`.

An empty list isn't allowed as `CommandNameAndArgs`, so `[]` should be rejected earlier.

> We could maybe address it by adding a check for empty arguments to MergeTool::external and removing the unwrap_or_else from MergeEditor::with_name, since I think that codepath never results in a valid tool.

That could be an option. I personally wouldn't refactor in that way, though, because I don't want to move the knowledge about `ExternalMergeTool::default()` to caller. The current implementation should work if we add/remove default `*_args`.

Reply:
OK, I won't think about it too hard then.